### PR TITLE
fix(core/automerge): care only about automergeSchedule when branch exists

### LINKED ISSUE #22436: Branch automerge skipped if out of schedule
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the bug

If a branch exists, and is configured for automerge, but is not in schedule, then the branch automerge will always be skipped.

The branch is skipped here: https://github.com/renovatebot/renovate/blob/e066924971bb0dc4ebfa0c95c919969c9eba7064/lib/workers/repository/update/branch/index.ts#L314-L320

Whereas branch automerge is not tried until hundreds of lines later: https://github.com/renovatebot/renovate/blob/e066924971bb0dc4ebfa0c95c919969c9eba7064/lib/workers/repository/update/branch/index.ts#L610-L614

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: branch.isModified() = false (repository=my-project/some-repo, branch=renovate/agent-images-automatically)
DEBUG: Checking schedule(before 8am, null) (repository=my-project/some-repo, branch=renovate/agent-images-automatically)
DEBUG: Checking 1 schedule(s) (repository=my-project/some-repo, branch=renovate/agent-images-automatically)
DEBUG: Checking schedule "before 8am" (repository=my-project/some-repo, branch=renovate/agent-images-automatically)
DEBUG: Package not scheduled (repository=my-project/some-repo, branch=renovate/agent-images-automatically)
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
When there is no pr and pr-creation is behind schedule, check to see if the branch is set up for automerge before skipping.

## Context
Closes https://github.com/renovatebot/renovate/issues/22436

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -190,6 +190,10 @@ So for example you could choose to automerge all (passing) `devDependencies` onl
 }
 ```
 
+<!-- prettier-ignore -->
+!!! note
+    Branch automerge ignores normal schedule and only respects [`automergeSchedule`](#automergeSchedule).
```
Comment (RahulGautamSingh) [inline docs/usage/configuration-options.md:195]:
```suggestion
    Branch automerge ignores normal schedule and only respects [`automergeSchedule`](#automergeschedule).
```

**Comment by RahulGautamSingh** [comment]
TestRun: https://github.com/RahulGautamSingh-testing/automerge-schedule

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -190,6 +190,10 @@ So for example you could choose to automerge all (passing) `devDependencies` onl
 }
 ```
 
+<!-- prettier-ignore -->
+!!! note
+    Branch automerge ignores normal schedule and only respects [`automergeSchedule`](#automergeschedule).
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:195]:
```suggestion
    Branch automerge ignores the normal schedule and only follows the [`automergeSchedule`](#automergeschedule).
```

File: lib/workers/repository/update/branch/index.ts [hunk 3]
```diff
@@ -310,7 +310,10 @@ export async function processBranch(
         };
       }
       // istanbul ignore if
-      if (!branchPr) {
+      if (
+        !branchPr &&
+        !(config.automerge && config.automergeType === 'branch') // if branch is configured for automerge no need of a PR
```
Comment (HonkingGoose) [inline lib/workers/repository/update/branch/index.ts:315]:
```suggestion
        !(config.automerge && config.automergeType === 'branch') // if branch is configured for automerge there's no need for a PR
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/workers/repository/update/branch/index.ts [hunk 4]
```diff
@@ -310,7 +310,10 @@ export async function processBranch(
         };
       }
       // istanbul ignore if
```
Comment (viceice) [inline lib/workers/repository/update/branch/index.ts:312]: is this still required?

**Review by viceice — COMMENTED** [review]
do we need to ensure we skip PR creation when branch automerge failed? 🤔

@rarkins WDYT?

**Comment by RahulGautamSingh** [comment]
> do we need to ensure we skip PR creation when branch automerge failed? 🤔

I thought this was already handled here. We create PR if automerge failed, so ensuring we skip this will change behaviour right?
https://github.com/renovatebot/renovate/blob/bbf4183b04aa123f242ad15757b16de2dc722431/lib/workers/repository/update/branch/index.ts#L640-L651

**Review by HonkingGoose — APPROVED** [review]
The docs look good to me. I'll leave the code review to the maintainers.

**Review by viceice — DISMISSED** [review]

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -190,6 +190,10 @@ So for example you could choose to automerge all (passing) `devDependencies` onl
 }
 ```
 
+<!-- prettier-ignore -->
+!!! note
+    Branch automerge ignores the normal schedule and only follows the [`automergeSchedule`](#automergeschedule).
```
Comment (rarkins) [inline docs/usage/configuration-options.md:195]: Are you sure we don't follow the `schedule` when it comes to _creating_ the branch to be automerged?

Comment (RahulGautamSingh) [reply docs/usage/configuration-options.md:195]: Branch creation still depends on `schedule`

Comment (RahulGautamSingh) [inline docs/usage/configuration-options.md:195]:
```suggestion
<!-- prettier-ignore -->
!!! note
    Branches creation follows [`schedule`](#schedule) and the automerge follows [`automergeSchedule`](#automergeschedule).
```

Comment (rarkins) [reply docs/usage/configuration-options.md:195]: I think this comment is potentially confusing

**Review by rarkins — APPROVED** [review]

**Pull request merged**